public/js/controllers/address_controller.js

```diff
@@ -175,21 +175,35 @@ export default class extends Controller {
     ctrl.bindElements()
     ctrl.bindEvents()
     ctrl.query = new TurboQuery()
-
+    ctrl.defaultSettings = {
+      chart: 'balance',
+      zoom: 'ijhhasg0-kbhgjkw0',
+      zoombutton: 'all',
+      bin: 'month',
+      flow: '',
+      n: '',
+      start: '',
+      txntype: ''
+    }
+    // ?chart=balance&zoom=ijhhasg0-kbhgjkw0&bin=month
     // These two are templates for query parameter sets.
     // When url query parameters are set, these will also be updated.
     var settings = ctrl.settings = TurboQuery.nullTemplate(['chart', 'zoom', 'bin', 'flow',
-      'n', 'start', 'txntype'])
+      'n', 'start', 'txntype', 'zoombutton'])
 
     ctrl.state = Object.assign({}, settings)
 
     // Get initial view settings from the url
     ctrl.query.update(settings)
     ctrl.setChartType()
     if (settings.flow) ctrl.setFlowChecks()
-    if (settings.zoom !== null) {
+    if (settings.zoombutton !== null) {
       ctrl.zoomButtons.forEach((button) => {
-        button.classList.remove('btn-selected')
+        if (button.name !== settings.zoombutton) {
```

Comment:
Can we use the convention of setting a `data-option` on the zoom buttons and decoding the `settings.zoom` to highlight the selected button?